ui/pages/confirmations/components/confirm/info/approve/hooks/use-approve-token-simulation.ts

```diff
@@ -18,34 +18,35 @@ export const useApproveTokenSimulation = (
   decimals: string,
 ) => {
   const locale = useSelector(getIntlLocale);
-
   const { isNFT, pending: isNFTPending } = useIsNFT(transactionMeta);
-
   const decodedResponse = useDecodedTransactionData();
-
   const { value, pending } = decodedResponse;
 
-  const decodedSpendingCap = value
-    ? new BigNumber(value.data[0].params[1].value)
-        .dividedBy(new BigNumber(10).pow(Number(decimals)))
-        .toNumber()
-    : 0;
-
-  const tokenPrefix = isNFT ? '#' : '';
-  const formattedTokenNum = isNFT
-    ? decodedSpendingCap
-    : new Intl.NumberFormat(locale).format(decodedSpendingCap);
-
-  let tokenAmount;
-  if (!isNFT && isSpendingCapUnlimited(decodedSpendingCap)) {
-    tokenAmount = UNLIMITED_MSG;
-  } else {
-    tokenAmount = `${tokenPrefix}${formattedTokenNum}`;
-  }
+  const decodedSpendingCap = useMemo(() => {
+    return value
+      ? new BigNumber(value.data[0].params[1].value)
+          .dividedBy(new BigNumber(10).pow(Number(decimals)))
+          .toNumber()
+      : 0;
```

Comment:
similar to https://github.com/MetaMask/metamask-extension/pull/26845#discussion_r1754570299
 
 we can use `calcTokenAmount(value.data[0].params[1].value, decimals).toFixed()` here
```suggestion
    return value
      ? calcTokenAmount(value.data[0].params[1].value, decimals).toFixed()
      : 0;
```